compaction/compaction_manager.cc

```diff
@@ -277,16 +279,17 @@ compaction_manager::task::task(compaction_manager& mgr, compaction::table_state*
     , _description(std::move(desc))
 {}
 
-future<> compaction_manager::perform_task(shared_ptr<compaction_manager::task> task) {
+future<compaction_manager::task::validation_errors_stats> compaction_manager::perform_task(shared_ptr<compaction_manager::task> task) {
```

Comment:
I hope that by storing the validation counters in `compaction_data` (similar to `total_partitions` and `total_keys_written`), we can copy them to `compaction_result` in `compaction::finish()`, and we can return it here.
It makes more sense to return a generic `compaction_result`, then scrub specific stats.

Reply:
If we returned `compaction_result` from each function up to `perform_task`, then we would need to go through `do_run`:
https://github.com/scylladb/scylla/blob/d566755234db2f7754a7ca0fa49e390ab1319836/compaction/compaction_manager.cc#L1262-L1272
and thus we would need to gather the returned value from each `validate_sstable`. Do we want to join `new_sstables` vectors and pass it through many functions, while we do not really use it anyhow?

I was thinking about creating a `compaction_counters` struct with `fixed_validation_errors` and `validation_errors` (and possibly some others), making the field of that type inside `compaction_result`. That would be effectively equal to renaming `compaction_task_stats` (and `validation_errors_stats`), though.

What do you think about it?

Reply:
Sounds good, but I'll need to see the patch to understand it better.